

<!DOCTYPE html>
<!--[if IE 8]><html class="no-js lt-ie9" lang="en" > <![endif]-->
<!--[if gt IE 8]><!--> <html class="no-js" lang="en" > <!--<![endif]-->
<head>
  <meta charset="utf-8">
  
  <meta name="viewport" content="width=device-width, initial-scale=1.0">
  
  <title>Transcriptome analysis of coral &amp; zooxanthellae &mdash; Coral multi-omics 1.0 documentation</title>
  

  
  

  

  
  
    

  

  
  
    <link rel="stylesheet" href="_static/main.css" type="text/css" />
  

  

  
        <link rel="index" title="Index"
              href="genindex.html"/>
        <link rel="search" title="Search" href="search.html"/>
    <link rel="top" title="Coral multi-omics 1.0 documentation" href="index.html"/>
        <link rel="next" title="Taxonomic assignments of the contigs" href="RNAseq/Contig_taxonomy.html"/>
        <link rel="prev" title="UniFrac analysis" href="16S/UniFrac.html"/> 

  
  <script src="_static/js/modernizr.min.js"></script>

</head>

<body class="wy-body-for-nav" role="document">

  <div class="wy-grid-for-nav">

    
    <nav data-toggle="wy-nav-shift" class="wy-nav-side">
      <div class="wy-side-scroll">
        <div class="wy-side-nav-search">
          

          
            <a href="index.html" class="icon icon-home"> Coral multi-omics
          

          
          </a>

          
            
            
              <div class="version">
                1.0
              </div>
            
          

          
<div role="search">
  <form id="rtd-search-form" class="wy-form" action="search.html" method="get">
    <input type="text" name="q" placeholder="Search docs" />
    <input type="hidden" name="check_keywords" value="yes" />
    <input type="hidden" name="area" value="default" />
  </form>
</div>

          
        </div>

        <div class="wy-menu wy-menu-vertical" data-spy="affix" role="navigation" aria-label="main navigation">
          
            
            
                <ul class="current">
<li class="toctree-l1"><a class="reference internal" href="16S.html">16S rRNA amplicon-seq analysis</a></li>
<li class="toctree-l1 current"><a class="current reference internal" href="#">Transcriptome analysis of coral &amp; zooxanthellae</a><ul>
<li class="toctree-l2"><a class="reference internal" href="#data-preprocessing">Data preprocessing</a><ul>
<li class="toctree-l3"><a class="reference internal" href="#quality-check">Quality check</a></li>
<li class="toctree-l3"><a class="reference internal" href="#preprocessing">Preprocessing</a></li>
</ul>
</li>
<li class="toctree-l2"><a class="reference internal" href="#transcriptome-profiling-of-corals">Transcriptome profiling of corals</a><ul>
<li class="toctree-l3"><a class="reference internal" href="#mapping-to-coral-genome">Mapping to coral genome</a></li>
<li class="toctree-l3"><a class="reference internal" href="#transcript-assembly">Transcript assembly</a></li>
<li class="toctree-l3"><a class="reference internal" href="#merge-gtf-files">Merge GTF files</a></li>
<li class="toctree-l3"><a class="reference internal" href="#calculate-expression-level">Calculate expression level</a></li>
</ul>
</li>
<li class="toctree-l2"><a class="reference internal" href="#transcriptome-profiling-of-zooxanthellae">Transcriptome profiling of zooxanthellae</a><ul>
<li class="toctree-l3"><a class="reference internal" href="#random-sampling-of-unmapped-reads">Random sampling of unmapped reads</a></li>
<li class="toctree-l3"><a class="reference internal" href="#de-novo-transcriptome-assembly">De novo transcriptome assembly</a></li>
<li class="toctree-l3"><a class="reference internal" href="#selection-of-zooxanthellae-transcripts">Selection of zooxanthellae transcripts</a></li>
<li class="toctree-l3"><a class="reference internal" href="#mapping-on-the-contigs">Mapping on the contigs</a></li>
<li class="toctree-l3"><a class="reference internal" href="#contig-clustering">Contig clustering</a></li>
<li class="toctree-l3"><a class="reference internal" href="#id1">Calculate expression level</a></li>
</ul>
</li>
<li class="toctree-l2"><a class="reference internal" href="#gene-annotation">Gene annotation</a><ul>
<li class="toctree-l3"><a class="reference internal" href="#kaas">KAAS</a></li>
<li class="toctree-l3"><a class="reference internal" href="#swissprot">SwissProt</a></li>
<li class="toctree-l3"><a class="reference internal" href="#ncbi-nr">NCBI nr</a></li>
</ul>
</li>
<li class="toctree-l2"><a class="reference internal" href="#downstream-analysis">Downstream analysis</a><ul>
<li class="toctree-l3"><a class="reference internal" href="#basic-analysis">Basic analysis</a><ul>
<li class="toctree-l4"><a class="reference internal" href="RNAseq/Contig_taxonomy.html">Taxonomic assignments of the contigs</a></li>
<li class="toctree-l4"><a class="reference internal" href="RNAseq/Mapping_rate.html">Mapping rate</a></li>
<li class="toctree-l4"><a class="reference internal" href="RNAseq/Correlation.html">Correlation patterns</a></li>
</ul>
</li>
<li class="toctree-l3"><a class="reference internal" href="#co-expression-network-analysis">Co-expression network analysis</a><ul>
<li class="toctree-l4"><a class="reference internal" href="RNAseq/WGCNA_power_estimation.html">Selection of optimal parameters</a></li>
<li class="toctree-l4"><a class="reference internal" href="RNAseq/WGCNA_module_detection.html">Module detection</a></li>
<li class="toctree-l4"><a class="reference internal" href="RNAseq/WGCNA_module_eigengenes.html">Identificaiton of Time- &amp; Colony-dependent modules</a></li>
<li class="toctree-l4"><a class="reference internal" href="RNAseq/WGCNA_module_eigengene_environment.html">Correlation of modules with environmental paramters</a></li>
<li class="toctree-l4"><a class="reference internal" href="RNAseq/CrossSpeciesNetwork.html">Construct cross-species network</a></li>
<li class="toctree-l4"><a class="reference internal" href="RNAseq/GeneSetAnalysis.html">Gene set analysis for the WGCNA modules</a></li>
</ul>
</li>
<li class="toctree-l3"><a class="reference internal" href="#differential-expression-analysis">Differential expression analysis</a><ul>
<li class="toctree-l4"><a class="reference internal" href="RNAseq/Differential_expression_analysis.html">Differential expression analysis</a></li>
</ul>
</li>
</ul>
</li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="Genome.html">Genome analysis of Endozoicomonas species</a></li>
<li class="toctree-l1"><a class="reference internal" href="Software.html">Software</a></li>
</ul>

            
          
        </div>
      </div>
    </nav>

    <section data-toggle="wy-nav-shift" class="wy-nav-content-wrap">

      
      <nav class="wy-nav-top" role="navigation" aria-label="top navigation">
        <i data-toggle="wy-nav-top" class="fa fa-bars"></i>
        <a href="index.html">Coral multi-omics</a>
      </nav>


      
      <div class="wy-nav-content">
        <div class="rst-content">
          

 



<div role="navigation" aria-label="breadcrumbs navigation">
  <ul class="wy-breadcrumbs">
    <li><a href="index.html">Docs</a> &raquo;</li>
      
    <li>Transcriptome analysis of coral &amp; zooxanthellae</li>
      <li class="wy-breadcrumbs-aside">
        
          
            <a href="_sources/RNAseq.rst.txt" rel="nofollow"> View page source</a>
          
        
      </li>
  </ul>
  <hr/>
</div>
          <div role="main" class="document" itemscope="itemscope" itemtype="http://schema.org/Article">
           <div itemprop="articleBody">
            
  <div class="section" id="transcriptome-analysis-of-coral-zooxanthellae">
<h1>Transcriptome analysis of coral &amp; zooxanthellae<a class="headerlink" href="#transcriptome-analysis-of-coral-zooxanthellae" title="Permalink to this headline">¶</a></h1>
<div class="section" id="data-preprocessing">
<h2>Data preprocessing<a class="headerlink" href="#data-preprocessing" title="Permalink to this headline">¶</a></h2>
<div class="section" id="quality-check">
<h3>Quality check<a class="headerlink" href="#quality-check" title="Permalink to this headline">¶</a></h3>
<p>Quality check was performed by FASTQC.</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre>fastqc -o <span class="si">${</span><span class="nv">outdir</span><span class="si">}</span> <span class="si">${</span><span class="nv">outdir</span><span class="si">}</span>/<span class="si">${</span><span class="nv">prefix</span><span class="si">}</span>_R1.fastq
fastqc -o <span class="si">${</span><span class="nv">outdir</span><span class="si">}</span> <span class="si">${</span><span class="nv">outdir</span><span class="si">}</span>/<span class="si">${</span><span class="nv">prefix</span><span class="si">}</span>_R2.fastq
</pre></div>
</div>
</div>
<div class="section" id="preprocessing">
<h3>Preprocessing<a class="headerlink" href="#preprocessing" title="Permalink to this headline">¶</a></h3>
<p>FASTQ files were preprocessed by bbduk in BBtools.</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre>bbduk -Xmx1g <span class="se">\</span>
   <span class="nv">in1</span><span class="o">=</span><span class="si">${</span><span class="nv">indir</span><span class="si">}</span>/<span class="si">${</span><span class="nv">prefix</span><span class="si">}</span>_R1.fastq <span class="se">\</span>
   <span class="nv">in2</span><span class="o">=</span><span class="si">${</span><span class="nv">indir</span><span class="si">}</span>/<span class="si">${</span><span class="nv">prefix</span><span class="si">}</span>_R2.fastq <span class="se">\</span>
   <span class="nv">out1</span><span class="o">=</span><span class="si">${</span><span class="nv">outdir</span><span class="si">}</span>/<span class="si">${</span><span class="nv">prefix</span><span class="si">}</span>_R1.fastq <span class="se">\</span>
   <span class="nv">out2</span><span class="o">=</span><span class="si">${</span><span class="nv">outdir</span><span class="si">}</span>/<span class="si">${</span><span class="nv">prefix</span><span class="si">}</span>_R2.fastq <span class="se">\</span>
   <span class="nv">ref</span><span class="o">=</span><span class="si">${</span><span class="nv">adapter</span><span class="si">}</span> <span class="se">\</span>
   <span class="nv">ktrim</span><span class="o">=</span>r <span class="se">\ </span><span class="c1"># Trim adapter from 3&#39; end (as usual adapter trimming</span>
   <span class="nv">k</span><span class="o">=</span><span class="m">23</span> <span class="se">\ </span><span class="c1"># k-mer size to use</span>
   <span class="nv">mink</span><span class="o">=</span><span class="m">11</span> <span class="se">\ </span><span class="c1"># Minimum match size</span>
   <span class="nv">hdist</span><span class="o">=</span><span class="m">1</span> <span class="se">\ </span><span class="c1"># Hamming distance: Allows 1 mismatch</span>
   tpe <span class="se">\ </span><span class="c1"># Trim adapters based on pair overlap detection using BBMerge</span>
   tbo <span class="se">\ </span><span class="c1"># Trim both reads to the same length</span>
   <span class="nv">qtrim</span><span class="o">=</span>rl <span class="nv">trimq</span><span class="o">=</span><span class="m">10</span> <span class="se">\ </span><span class="c1"># Remove low-quality bases (QC&lt;10) from both sides</span>
   <span class="nv">maq</span><span class="o">=</span><span class="m">10</span> <span class="c1"># Remove low-quality reads (average QC &lt; 20)</span>
</pre></div>
</div>
</div>
</div>
<div class="section" id="transcriptome-profiling-of-corals">
<h2>Transcriptome profiling of corals<a class="headerlink" href="#transcriptome-profiling-of-corals" title="Permalink to this headline">¶</a></h2>
<div class="section" id="mapping-to-coral-genome">
<h3>Mapping to coral genome<a class="headerlink" href="#mapping-to-coral-genome" title="Permalink to this headline">¶</a></h3>
<p>Sequence reads were mapped onto reference genome of Acropora tenuis by HISAT2.</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre>hisat2 <span class="se">\</span>
   -x <span class="si">${</span><span class="nv">reference</span><span class="si">}</span> <span class="se">\ </span><span class="c1"># Index generated from reference genome</span>
   -1 <span class="si">${</span><span class="nv">directory_preprocessed_fastq</span><span class="si">}</span>/<span class="si">${</span><span class="nv">prefix</span><span class="si">}</span>_R1.fastq <span class="se">\ </span><span class="c1"># Input: Read 1</span>
   -2 <span class="si">${</span><span class="nv">directory_preprocessed_fastq</span><span class="si">}</span>/<span class="si">${</span><span class="nv">prefix</span><span class="si">}</span>_R2.fastq <span class="se">\ </span><span class="c1"># Input: Read 2</span>
   -p <span class="m">1</span> <span class="se">\ </span><span class="c1"># Thread number</span>
   -S <span class="si">${</span><span class="nv">directory_mapping</span><span class="si">}</span>/<span class="si">${</span><span class="nv">prefix</span><span class="si">}</span>.Acropora_tenuis_UT.sam <span class="se">\ </span><span class="c1"># Output: SAM format</span>
   --un-conc <span class="si">${</span><span class="nv">directory_mapping</span><span class="si">}</span>/<span class="si">${</span><span class="nv">prefix</span><span class="si">}</span>.Acropora_tenuis_UT.unmapped.R%.fastq <span class="c1"># Output: unmapped reads, used for transcriptome assembly in following section</span>
   --dta <span class="c1"># This option is required for StringTie</span>
</pre></div>
</div>
</div>
<div class="section" id="transcript-assembly">
<h3>Transcript assembly<a class="headerlink" href="#transcript-assembly" title="Permalink to this headline">¶</a></h3>
<p>Mapping results were used for transcriptome assembly by StringTie.</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span class="c1"># convert to BAM and sort, necessary for StringTie</span>
samtools sort <span class="se">\</span>
   -O bam <span class="se">\</span>
   -o <span class="si">${</span><span class="nv">directory_mapping</span><span class="si">}</span>/<span class="si">${</span><span class="nv">prefix</span><span class="si">}</span>.Acropora_tenuis_UT.bam <span class="se">\</span>
   <span class="si">${</span><span class="nv">directory_mapping</span><span class="si">}</span>/<span class="si">${</span><span class="nv">prefix</span><span class="si">}</span>.Acropora_tenuis_UT.sam
rm <span class="si">${</span><span class="nv">directory_mapping</span><span class="si">}</span>/<span class="si">${</span><span class="nv">prefix</span><span class="si">}</span>.Acropora_tenuis_UT.sam

<span class="c1"># StringTie and create new GTF file</span>
stringtie <span class="se">\</span>
   <span class="si">${</span><span class="nv">directory_mapping</span><span class="si">}</span>/<span class="si">${</span><span class="nv">prefix</span><span class="si">}</span>.Acropora_tenuis_UT.bam <span class="se">\</span>
   -G <span class="si">${</span><span class="nv">gtf_file</span><span class="si">}</span> <span class="se">\</span>
   -A <span class="si">${</span><span class="nv">directory_mapping</span><span class="si">}</span>/<span class="si">${</span><span class="nv">prefix</span><span class="si">}</span>.Acropora_tenuis_UT.StringTie.tab <span class="se">\</span>
   -o <span class="si">${</span><span class="nv">directory_mapping</span><span class="si">}</span>/<span class="si">${</span><span class="nv">prefix</span><span class="si">}</span>.Acropora_tenuis_UT.StringTie.gtf
</pre></div>
</div>
</div>
<div class="section" id="merge-gtf-files">
<h3>Merge GTF files<a class="headerlink" href="#merge-gtf-files" title="Permalink to this headline">¶</a></h3>
<p>The above step generates GTF files for all samples. This step merges all files and create one representative file.</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span class="c1"># Create list of GFF</span>
ls <span class="si">${</span><span class="nv">directory_mapping</span><span class="si">}</span>/*gtf &gt; <span class="si">${</span><span class="nv">merged_gtf_file</span><span class="si">}</span>

<span class="c1"># Merge GTF files</span>
stringtie --merge <span class="se">\</span>
   -p <span class="m">1</span> <span class="se">\ </span><span class="c1"># Thread</span>
   -G <span class="si">${</span><span class="nv">merged_gtf_file</span><span class="si">}</span> <span class="se">\ </span><span class="c1"># List of GTF files</span>
   -o <span class="si">${</span><span class="nv">directory_mapping</span><span class="si">}</span>/stringtie_mergelist.txt <span class="c1"># Output</span>
</pre></div>
</div>
</div>
<div class="section" id="calculate-expression-level">
<h3>Calculate expression level<a class="headerlink" href="#calculate-expression-level" title="Permalink to this headline">¶</a></h3>
<p>Gene expression levels were estimated with the merged GTF file by StringTie.</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span class="c1"># FPKM and TPM</span>
stringtie <span class="se">\</span>
   <span class="si">${</span><span class="nv">outdir</span><span class="si">}</span>/<span class="si">${</span><span class="nv">prefix</span><span class="si">}</span>.Acropora_tenuis_UT.bam <span class="se">\</span>
   -e -G <span class="si">${</span><span class="nv">merged_gtf_file</span><span class="si">}</span> <span class="se">\</span>
   -A <span class="si">${</span><span class="nv">directory_mapping</span><span class="si">}</span>/<span class="si">${</span><span class="nv">prefix</span><span class="si">}</span>.Acropora_tenuis_UT.StringTie.Merged.tab
   -o <span class="si">${</span><span class="nv">directory_mapping</span><span class="si">}</span>/<span class="si">${</span><span class="nv">prefix</span><span class="si">}</span>.Acropora_tenuis_UT.StringTie.Merged.gtf

<span class="c1"># Count table</span>
prepDE.py <span class="se">\</span>
   -i list.gtf.UT <span class="se">\</span>
   -g Acropora_tenuis_UT.HISAT2.StringTie.Count.txt
</pre></div>
</div>
</div>
</div>
<div class="section" id="transcriptome-profiling-of-zooxanthellae">
<h2>Transcriptome profiling of zooxanthellae<a class="headerlink" href="#transcriptome-profiling-of-zooxanthellae" title="Permalink to this headline">¶</a></h2>
<div class="section" id="random-sampling-of-unmapped-reads">
<h3>Random sampling of unmapped reads<a class="headerlink" href="#random-sampling-of-unmapped-reads" title="Permalink to this headline">¶</a></h3>
<p>From unmapped reads of each sample in section <cite>1. Mapping to coral genome</cite>, 1 million reads were randomly selected by seqtk.</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre>seqtk sample -s100 <span class="si">${</span><span class="nv">indir</span><span class="si">}</span>/<span class="si">${</span><span class="nv">prefix</span><span class="si">}</span>.Acropora_tenuis_UT.unmapped.R1.fastq.gz <span class="m">1000000</span> &gt; <span class="si">${</span><span class="nv">indir2</span><span class="si">}</span>/<span class="si">${</span><span class="nv">prefix</span><span class="si">}</span>.Acropora_tenuis_UT.unmapped.R1.fastq
seqtk sample -s100 <span class="si">${</span><span class="nv">indir</span><span class="si">}</span>/<span class="si">${</span><span class="nv">prefix</span><span class="si">}</span>.Acropora_tenuis_UT.unmapped.R2.fastq.gz <span class="m">1000000</span> &gt; <span class="si">${</span><span class="nv">indir2</span><span class="si">}</span>/<span class="si">${</span><span class="nv">prefix</span><span class="si">}</span>.Acropora_tenuis_UT.unmapped.R2.fastq
</pre></div>
</div>
</div>
<div class="section" id="de-novo-transcriptome-assembly">
<h3>De novo transcriptome assembly<a class="headerlink" href="#de-novo-transcriptome-assembly" title="Permalink to this headline">¶</a></h3>
<p>De novo transcriptome assembly was performed by rnaSPAdes.</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre>rnaspades <span class="se">\</span>
   -o <span class="si">${</span><span class="nv">output</span><span class="si">}</span> <span class="se">\ </span><span class="c1"># Output folder</span>
   -t <span class="m">8</span> <span class="se">\ </span><span class="c1"># Thread</span>
   -m <span class="m">128</span> <span class="se">\ </span><span class="c1"># Memory required</span>
   --dataset rnaSPAdes.sample_file.CREST_v1.1Mreads.UT.yaml <span class="c1"># Input file list</span>
</pre></div>
</div>
</div>
<div class="section" id="selection-of-zooxanthellae-transcripts">
<h3>Selection of zooxanthellae transcripts<a class="headerlink" href="#selection-of-zooxanthellae-transcripts" title="Permalink to this headline">¶</a></h3>
<p>Select contigs derived from zooxanthellae by similarity search against protein sequences of coral &amp; zooxanthellae.</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre>blastx <span class="se">\</span>
   -query <span class="si">${</span><span class="nv">output</span><span class="si">}</span>/transcripts.fasta <span class="se">\</span>
   -db <span class="si">${</span><span class="nv">db</span><span class="si">}</span> <span class="se">\</span>
   -out <span class="si">${</span><span class="nv">input</span><span class="si">}</span>.Coral_Zooxanthellae_Classify.txt -outfmt <span class="m">6</span> -num_threads 10
</pre></div>
</div>
</div>
<div class="section" id="mapping-on-the-contigs">
<h3>Mapping on the contigs<a class="headerlink" href="#mapping-on-the-contigs" title="Permalink to this headline">¶</a></h3>
<p>Mapping of the unmapped reads on the contigs was performed by Salmon.</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre>salmon quant <span class="se">\</span>
   --index <span class="si">${</span><span class="nv">reference</span><span class="si">}</span> <span class="se">\ </span><span class="c1"># Index file created from ${output}/transcripts.zooxanthellae.fasta</span>
   --libType IU <span class="se">\</span>
   --dumpEq <span class="se">\</span>
   -1 <span class="si">${</span><span class="nv">indir2</span><span class="si">}</span>/<span class="si">${</span><span class="nv">prefix</span><span class="si">}</span>.Acropora_tenuis_UT.unmapped.R1.fastq <span class="se">\</span>
   -2 <span class="si">${</span><span class="nv">indir2</span><span class="si">}</span>/<span class="si">${</span><span class="nv">prefix</span><span class="si">}</span>.Acropora_tenuis_UT.unmapped.R2.fastq <span class="se">\</span>
   --output <span class="si">${</span><span class="nv">outdir</span><span class="si">}</span>/<span class="si">${</span><span class="nv">prefix</span><span class="si">}</span>.salmon
</pre></div>
</div>
</div>
<div class="section" id="contig-clustering">
<h3>Contig clustering<a class="headerlink" href="#contig-clustering" title="Permalink to this headline">¶</a></h3>
<p>Re-clusteirng of contigs were performed by Corset.</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre>corset <span class="se">\</span>
   -f <span class="nb">true</span> <span class="se">\</span>
   -i salmon_eq_classes <span class="se">\</span>
   -p corset_salmon <span class="se">\</span>
   -g 1,1,1,2,2,2,3,3,3,4,4,4,5,5,5,6,6,6,7,7,7,8,8,8,9,9,9,10,10,10,11,11,11,12,12,12,13,13,13,14,14,14,15,15,15,16,16,16,17,17,17,18,18,18,19,19,19,20,20,20,21,21,21 <span class="se">\</span>
   -n 201411_Is1,201411_Is2,201411_Is3,201411_Ky1,201411_Ky2,201411_Ky3,201411_Mi1,201411_Mi2,201411_Mi3,201502_Is1,201502_Is2,201502_Is3,201502_Ky1,201502_Ky2,201502_Ky3,201502_Mi1,201502_Mi2,201502_Mi3,201505_Is1,201505_Is2,201505_Is3,201505_Mi1,201505_Mi2,201505_Mi3,201508_Is1,201508_Is2,201508_Is3,201508_Mi1,201508_Mi2,201508_Mi3,201511_Is1,201511_Is2,201511_Is3,201511_Mi1,201511_Mi2,201511_Mi3,201602_Is1,201602_Is2,201602_Is3,201602_Mi1,201602_Mi2,201602_Mi3,201603_Is1,201603_Is2,201603_Is3,201603_Mi1,201603_Mi2,201603_Mi3,201605_Is1,201605_Is2,201605_Is3,201605_Mi1,201605_Mi2,201605_Mi3,201608_Is1,201608_Is2,201608_Is3,201608_Mi1,201608_Mi2,201608_Mi3,201609_Is1,201609_Is2,201609_Is3 <span class="se">\</span>
   201411_Is1.salmon/aux_info/eq_classes.txt 201411_Is2.salmon/aux_info/eq_classes.txt 201411_Is3.salmon/aux_info/eq_classes.txt 201411_Ky1.salmon/aux_info/eq_classes.txt 201411_Ky2.salmon/aux_info/eq_classes.txt 201411_Ky3.salmon/aux_info/eq_classes.txt 201411_Mi1.salmon/aux_info/eq_classes.txt 201411_Mi2.salmon/aux_info/eq_classes.txt 201411_Mi3.salmon/aux_info/eq_classes.txt 201502_Is1.salmon/aux_info/eq_classes.txt 201502_Is2.salmon/aux_info/eq_classes.txt 201502_Is3.salmon/aux_info/eq_classes.txt 201502_Ky1.salmon/aux_info/eq_classes.txt 201502_Ky2.salmon/aux_info/eq_classes.txt 201502_Ky3.salmon/aux_info/eq_classes.txt 201502_Mi1.salmon/aux_info/eq_classes.txt 201502_Mi2.salmon/aux_info/eq_classes.txt 201502_Mi3.salmon/aux_info/eq_classes.txt 201505_Is1.salmon/aux_info/eq_classes.txt 201505_Is2.salmon/aux_info/eq_classes.txt 201505_Is3.salmon/aux_info/eq_classes.txt 201505_Mi1.salmon/aux_info/eq_classes.txt 201505_Mi2.salmon/aux_info/eq_classes.txt 201505_Mi3.salmon/aux_info/eq_classes.txt 201508_Is1.salmon/aux_info/eq_classes.txt 201508_Is2.salmon/aux_info/eq_classes.txt 201508_Is3.salmon/aux_info/eq_classes.txt 201508_Mi1.salmon/aux_info/eq_classes.txt 201508_Mi2.salmon/aux_info/eq_classes.txt 201508_Mi3.salmon/aux_info/eq_classes.txt 201511_Is1.salmon/aux_info/eq_classes.txt 201511_Is2.salmon/aux_info/eq_classes.txt 201511_Is3.salmon/aux_info/eq_classes.txt 201511_Mi1.salmon/aux_info/eq_classes.txt 201511_Mi2.salmon/aux_info/eq_classes.txt 201511_Mi3.salmon/aux_info/eq_classes.txt 201602_Is1.salmon/aux_info/eq_classes.txt 201602_Is2.salmon/aux_info/eq_classes.txt 201602_Is3.salmon/aux_info/eq_classes.txt 201602_Mi1.salmon/aux_info/eq_classes.txt 201602_Mi2.salmon/aux_info/eq_classes.txt 201602_Mi3.salmon/aux_info/eq_classes.txt 201603_Is1.salmon/aux_info/eq_classes.txt 201603_Is2.salmon/aux_info/eq_classes.txt 201603_Is3.salmon/aux_info/eq_classes.txt 201603_Mi1.salmon/aux_info/eq_classes.txt 201603_Mi2.salmon/aux_info/eq_classes.txt 201603_Mi3.salmon/aux_info/eq_classes.txt 201605_Is1.salmon/aux_info/eq_classes.txt 201605_Is2.salmon/aux_info/eq_classes.txt 201605_Is3.salmon/aux_info/eq_classes.txt 201605_Mi1.salmon/aux_info/eq_classes.txt 201605_Mi2.salmon/aux_info/eq_classes.txt 201605_Mi3.salmon/aux_info/eq_classes.txt 201608_Is1.salmon/aux_info/eq_classes.txt 201608_Is2.salmon/aux_info/eq_classes.txt 201608_Is3.salmon/aux_info/eq_classes.txt 201608_Mi1.salmon/aux_info/eq_classes.txt 201608_Mi2.salmon/aux_info/eq_classes.txt 201608_Mi3.salmon/aux_info/eq_classes.txt 201609_Is1.salmon/aux_info/eq_classes.txt 201609_Is2.salmon/aux_info/eq_classes.txt 201609_Is3.salmon/aux_info/eq_classes.txt
</pre></div>
</div>
</div>
<div class="section" id="id1">
<h3>Calculate expression level<a class="headerlink" href="#id1" title="Permalink to this headline">¶</a></h3>
<p>Gene expression level was estimated with by RSEM.</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span class="c1"># Create index</span>
rsem-prepare-reference <span class="se">\</span>
   --num-threads <span class="m">1</span> <span class="se">\</span>
   --transcript-to-gene-map corset-clusters.txt <span class="se">\</span>
   --bowtie --bowtie-path <span class="si">${</span><span class="nv">bowtie</span><span class="si">}</span> <span class="se">\</span>
   <span class="si">${</span><span class="nv">output</span><span class="si">}</span>/transcripts.zooxanthellae.fasta <span class="se">\</span>
   <span class="si">${</span><span class="nv">rsem_reference</span><span class="si">}</span>

<span class="c1"># Gene expression profiling</span>
rsem <span class="se">\</span>
   --num-threads <span class="m">1</span> <span class="se">\</span>
   --bowtie-path <span class="si">${</span><span class="nv">bowtie</span><span class="si">}</span> <span class="se">\</span>
   --paired-end <span class="si">${</span><span class="nv">indir2</span><span class="si">}</span>/<span class="si">${</span><span class="nv">prefix</span><span class="si">}</span>.Acropora_tenuis_UT.unmapped.R1.fastq <span class="si">${</span><span class="nv">indir2</span><span class="si">}</span>/<span class="si">${</span><span class="nv">prefix</span><span class="si">}</span>.Acropora_tenuis_UT.unmapped.R2.fastq
   <span class="si">${</span><span class="nv">rsem_reference</span><span class="si">}</span> <span class="se">\</span>
   <span class="si">${</span><span class="nv">prefix</span><span class="si">}</span>.corset_salmon
</pre></div>
</div>
</div>
</div>
<div class="section" id="gene-annotation">
<h2>Gene annotation<a class="headerlink" href="#gene-annotation" title="Permalink to this headline">¶</a></h2>
<div class="section" id="kaas">
<h3>KAAS<a class="headerlink" href="#kaas" title="Permalink to this headline">¶</a></h3>
<p>Gene annotation with KEGG database was performed by KEGG Automatic Annotation Server (<a class="reference external" href="https://www.genome.jp/tools/kaas/">https://www.genome.jp/tools/kaas/</a>).</p>
</div>
<div class="section" id="swissprot">
<h3>SwissProt<a class="headerlink" href="#swissprot" title="Permalink to this headline">¶</a></h3>
<ol class="arabic">
<li><p class="first"><strong>Create ID mapping file</strong></p>
<div class="highlight-bash notranslate"><div class="highlight"><pre>wget ftp://ftp.uniprot.org/pub/databases/uniprot/current_release/knowledgebase/idmapping/idmapping_selected.tab.gz
gunzip ftp://ftp.uniprot.org/pub/databases/uniprot/current_release/knowledgebase/idmapping/idmapping_selected.tab.gz
cat idmapping_selected.tab <span class="p">|</span> awk <span class="s1">&#39;{split($0,a,&quot;\t&quot;); print(a[1]&quot;\t&quot;a[7])}&#39;</span> &gt; tmp
grep -v <span class="s1">$&#39;\t$&#39;</span> tmp &gt; Uniprot2GO.txt
rm tmp
</pre></div>
</div>
</li>
<li><p class="first"><strong>BLAST</strong></p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span class="c1"># BLASTp similarity search with Swissprot</span>
blastp <span class="se">\</span>
   -query <span class="si">${</span><span class="nv">input</span><span class="si">}</span> <span class="se">\</span>
   -db <span class="si">${</span><span class="nv">db</span><span class="si">}</span> <span class="se">\</span>
   -out <span class="si">${</span><span class="nv">input</span><span class="si">}</span>.swissprot_blastp.txt <span class="se">\</span>
   -outfmt <span class="m">6</span> <span class="se">\</span>
   -num_threads 10
</pre></div>
</div>
</li>
<li><p class="first"><strong>Parse results</strong></p>
<div class="highlight-bash notranslate"><div class="highlight"><pre>python parse_swissprot.py <span class="se">\</span>
   -i <span class="si">${</span><span class="nv">input</span><span class="si">}</span>.swissprot_blastp.txt <span class="se">\</span>
   -o <span class="si">${</span><span class="nv">input</span><span class="si">}</span>.swissprot_blastp.parse.txt <span class="se">\</span>
   -threshold 1e-5
</pre></div>
</div>
<div class="highlight-python notranslate"><div class="highlight"><pre><span class="c1"># parse_swissprot.py</span>
<span class="kn">import</span> <span class="nn">os</span>
<span class="kn">import</span> <span class="nn">sys</span>
<span class="kn">import</span> <span class="nn">argparse</span>

<span class="n">parser</span> <span class="o">=</span> <span class="n">argparse</span><span class="o">.</span><span class="n">ArgumentParser</span><span class="p">()</span>
<span class="n">parser</span><span class="o">.</span><span class="n">add_argument</span><span class="p">(</span><span class="s2">&quot;-i&quot;</span><span class="p">,</span><span class="n">dest</span><span class="o">=</span><span class="s2">&quot;ifile&quot;</span><span class="p">,</span><span class="n">action</span><span class="o">=</span><span class="s2">&quot;store&quot;</span><span class="p">,</span><span class="n">default</span><span class="o">=</span><span class="s2">&quot;&quot;</span><span class="p">,</span><span class="n">help</span><span class="o">=</span><span class="s2">&quot;Path of input file&quot;</span><span class="p">)</span>
<span class="n">parser</span><span class="o">.</span><span class="n">add_argument</span><span class="p">(</span><span class="s2">&quot;-o&quot;</span><span class="p">,</span><span class="n">dest</span><span class="o">=</span><span class="s2">&quot;ofile&quot;</span><span class="p">,</span><span class="n">action</span><span class="o">=</span><span class="s2">&quot;store&quot;</span><span class="p">,</span><span class="n">default</span><span class="o">=</span><span class="s2">&quot;&quot;</span><span class="p">,</span><span class="n">help</span><span class="o">=</span><span class="s2">&quot;Path of output file&quot;</span><span class="p">)</span>
<span class="n">parser</span><span class="o">.</span><span class="n">add_argument</span><span class="p">(</span><span class="s2">&quot;-threshold&quot;</span><span class="p">,</span><span class="n">dest</span><span class="o">=</span><span class="s2">&quot;threshold&quot;</span><span class="p">,</span><span class="n">action</span><span class="o">=</span><span class="s2">&quot;store&quot;</span><span class="p">,</span><span class="n">default</span><span class="o">=</span><span class="mf">1e-5</span><span class="p">,</span><span class="nb">type</span><span class="o">=</span><span class="nb">float</span><span class="p">,</span><span class="n">help</span><span class="o">=</span><span class="s2">&quot;e-value, threshold&quot;</span><span class="p">)</span>
<span class="n">parser</span><span class="o">.</span><span class="n">add_argument</span><span class="p">(</span><span class="s2">&quot;-threshold_col&quot;</span><span class="p">,</span><span class="n">dest</span><span class="o">=</span><span class="s2">&quot;threshold_col&quot;</span><span class="p">,</span><span class="n">action</span><span class="o">=</span><span class="s2">&quot;store&quot;</span><span class="p">,</span><span class="n">default</span><span class="o">=</span><span class="mi">11</span><span class="p">,</span><span class="nb">type</span><span class="o">=</span><span class="nb">int</span><span class="p">,</span><span class="n">help</span><span class="o">=</span><span class="s2">&quot;e-value, threshold_col&quot;</span><span class="p">)</span>
<span class="n">args</span> <span class="o">=</span> <span class="n">parser</span><span class="o">.</span><span class="n">parse_args</span><span class="p">()</span>

<span class="n">id2go</span> <span class="o">=</span> <span class="nb">dict</span><span class="p">()</span>
<span class="nb">file</span> <span class="o">=</span> <span class="s2">&quot;/archive/data/hgc0934/toru/CREST/Transcriptome.2020/DB/Uniprot/Uniprot2GO.txt&quot;</span>
<span class="n">fi</span> <span class="o">=</span> <span class="nb">open</span><span class="p">(</span><span class="nb">file</span><span class="p">)</span>
<span class="k">for</span> <span class="n">ii</span><span class="p">,</span><span class="n">jj</span> <span class="ow">in</span> <span class="nb">enumerate</span><span class="p">(</span><span class="n">fi</span><span class="p">):</span>
         <span class="n">data</span> <span class="o">=</span> <span class="n">jj</span><span class="o">.</span><span class="n">strip</span><span class="p">(</span><span class="s2">&quot;</span><span class="se">\n</span><span class="s2">&quot;</span><span class="p">)</span><span class="o">.</span><span class="n">split</span><span class="p">(</span><span class="s2">&quot;</span><span class="se">\t</span><span class="s2">&quot;</span><span class="p">)</span>
         <span class="n">go</span> <span class="o">=</span> <span class="n">data</span><span class="p">[</span><span class="mi">1</span><span class="p">]</span>
         <span class="n">id2go</span><span class="p">[</span><span class="n">data</span><span class="p">[</span><span class="mi">0</span><span class="p">]]</span> <span class="o">=</span> <span class="n">go</span>

<span class="nb">id</span> <span class="o">=</span> <span class="s2">&quot;&quot;</span>
<span class="n">ids</span> <span class="o">=</span> <span class="p">[]</span>
<span class="n">id2hit</span> <span class="o">=</span> <span class="nb">dict</span><span class="p">()</span>
<span class="n">index</span> <span class="o">=</span> <span class="n">args</span><span class="o">.</span><span class="n">threshold_col</span> <span class="o">-</span> <span class="mi">1</span>
<span class="k">with</span> <span class="nb">open</span><span class="p">(</span><span class="n">args</span><span class="o">.</span><span class="n">ifile</span><span class="p">)</span> <span class="k">as</span> <span class="n">fi</span><span class="p">:</span>
   <span class="k">for</span> <span class="n">ii</span><span class="p">,</span><span class="n">jj</span> <span class="ow">in</span> <span class="nb">enumerate</span><span class="p">(</span><span class="n">fi</span><span class="p">):</span>
         <span class="n">data</span> <span class="o">=</span> <span class="n">jj</span><span class="o">.</span><span class="n">strip</span><span class="p">(</span><span class="s2">&quot;</span><span class="se">\n</span><span class="s2">&quot;</span><span class="p">)</span><span class="o">.</span><span class="n">split</span><span class="p">(</span><span class="s2">&quot;</span><span class="se">\t</span><span class="s2">&quot;</span><span class="p">)</span>
         <span class="k">if</span> <span class="nb">id</span> <span class="o">!=</span> <span class="n">data</span><span class="p">[</span><span class="mi">0</span><span class="p">]:</span>
            <span class="n">evalue</span> <span class="o">=</span> <span class="n">data</span><span class="p">[</span><span class="n">index</span><span class="p">]</span>
            <span class="k">if</span> <span class="nb">float</span><span class="p">(</span><span class="n">evalue</span><span class="p">)</span> <span class="o">&lt;=</span> <span class="n">args</span><span class="o">.</span><span class="n">threshold</span><span class="p">:</span>
               <span class="n">uid</span> <span class="o">=</span> <span class="n">data</span><span class="p">[</span><span class="mi">1</span><span class="p">]</span><span class="o">.</span><span class="n">split</span><span class="p">(</span><span class="s2">&quot;|&quot;</span><span class="p">)[</span><span class="mi">1</span><span class="p">]</span>
               <span class="k">if</span> <span class="n">uid</span> <span class="ow">in</span> <span class="n">id2go</span><span class="p">:</span>
                  <span class="n">ids</span><span class="o">.</span><span class="n">append</span><span class="p">(</span><span class="n">data</span><span class="p">[</span><span class="mi">0</span><span class="p">])</span>
                  <span class="n">id2hit</span><span class="p">[</span><span class="n">data</span><span class="p">[</span><span class="mi">0</span><span class="p">]]</span> <span class="o">=</span> <span class="p">[</span><span class="n">uid</span><span class="p">,</span> <span class="n">evalue</span><span class="p">,</span> <span class="n">id2go</span><span class="p">[</span><span class="n">uid</span><span class="p">]]</span>
               <span class="k">else</span><span class="p">:</span>
                  <span class="n">ids</span><span class="o">.</span><span class="n">append</span><span class="p">(</span><span class="n">data</span><span class="p">[</span><span class="mi">0</span><span class="p">])</span>
                  <span class="n">id2hit</span><span class="p">[</span><span class="n">data</span><span class="p">[</span><span class="mi">0</span><span class="p">]]</span> <span class="o">=</span> <span class="p">[</span><span class="n">uid</span><span class="p">,</span> <span class="n">evalue</span><span class="p">,</span> <span class="s2">&quot;&quot;</span><span class="p">]</span>
            <span class="nb">id</span> <span class="o">=</span> <span class="n">data</span><span class="p">[</span><span class="mi">0</span><span class="p">]</span>

<span class="k">with</span> <span class="nb">open</span><span class="p">(</span><span class="n">args</span><span class="o">.</span><span class="n">ofile</span><span class="p">,</span><span class="s2">&quot;w&quot;</span><span class="p">)</span> <span class="k">as</span> <span class="n">fo</span><span class="p">:</span>
   <span class="k">for</span> <span class="nb">id</span> <span class="ow">in</span> <span class="n">ids</span><span class="p">:</span>
      <span class="n">fo</span><span class="o">.</span><span class="n">write</span><span class="p">(</span><span class="nb">id</span> <span class="o">+</span> <span class="s2">&quot;</span><span class="se">\t</span><span class="s2">&quot;</span> <span class="o">+</span> <span class="s2">&quot;</span><span class="se">\t</span><span class="s2">&quot;</span><span class="o">.</span><span class="n">join</span><span class="p">(</span><span class="n">id2hit</span><span class="p">[</span><span class="nb">id</span><span class="p">])</span> <span class="o">+</span> <span class="s2">&quot;</span><span class="se">\n</span><span class="s2">&quot;</span><span class="p">)</span>
<span class="n">fo</span><span class="o">.</span><span class="n">close</span><span class="p">()</span>
</pre></div>
</div>
</li>
</ol>
</div>
<div class="section" id="ncbi-nr">
<h3>NCBI nr<a class="headerlink" href="#ncbi-nr" title="Permalink to this headline">¶</a></h3>
<p>Similarity search against NCBI nr was performed to assign taxonomy and discriminate transcripts from coral &amp; zooxanthellae. All scripts used in this step can be downloaded from <a class="reference external" href="https://github.com/tmaruy/blast2taxonomy">https://github.com/tmaruy/blast2taxonomy</a>.</p>
<ol class="arabic">
<li><p class="first"><strong>Preparation of DB</strong></p>
<div class="highlight-bash notranslate"><div class="highlight"><pre>git clone https://github.com/tmaruy/blast2taxonomy.git
chmod +x preprare_taxdb.sh
./prepare_taxdb.sh
</pre></div>
</div>
</li>
<li><p class="first"><strong>Similarity search with Diamond</strong></p>
<div class="highlight-bash notranslate"><div class="highlight"><pre>diamond blastx <span class="se">\</span>
   -q <span class="si">${</span><span class="nv">input</span><span class="si">}</span> <span class="se">\</span>
   -d <span class="si">${</span><span class="nv">db</span><span class="si">}</span> <span class="se">\</span>
   -o <span class="si">${</span><span class="nv">input</span><span class="si">}</span>.nr_diamond.txt <span class="se">\</span>
   -f <span class="m">6</span> <span class="se">\</span>
   -p 8
</pre></div>
</div>
</li>
<li><p class="first"><strong>Parse results</strong></p>
<dl class="docutils">
<dt>python parse_nr.py </dt>
<dd><p class="first last">-i ${input}.nr_diamond.txt -o ${input}.nr_diamond.parse.txt -threshold 1e-5</p>
</dd>
</dl>
</li>
</ol>
</div>
</div>
<div class="section" id="downstream-analysis">
<h2>Downstream analysis<a class="headerlink" href="#downstream-analysis" title="Permalink to this headline">¶</a></h2>
<div class="section" id="basic-analysis">
<h3>Basic analysis<a class="headerlink" href="#basic-analysis" title="Permalink to this headline">¶</a></h3>
<div class="toctree-wrapper compound">
<ul>
<li class="toctree-l1"><a class="reference internal" href="RNAseq/Contig_taxonomy.html">Taxonomic assignments of the contigs</a></li>
<li class="toctree-l1"><a class="reference internal" href="RNAseq/Mapping_rate.html">Mapping rate</a></li>
<li class="toctree-l1"><a class="reference internal" href="RNAseq/Correlation.html">Correlation patterns</a></li>
</ul>
</div>
</div>
<div class="section" id="co-expression-network-analysis">
<h3>Co-expression network analysis<a class="headerlink" href="#co-expression-network-analysis" title="Permalink to this headline">¶</a></h3>
<p>Gene co-expression newtork was constructed with coral and zooxanthellae transcriptome respectively by an R package WGCNA.</p>
<div class="toctree-wrapper compound">
<ul>
<li class="toctree-l1"><a class="reference internal" href="RNAseq/WGCNA_power_estimation.html">Selection of optimal parameters</a></li>
<li class="toctree-l1"><a class="reference internal" href="RNAseq/WGCNA_module_detection.html">Module detection</a></li>
<li class="toctree-l1"><a class="reference internal" href="RNAseq/WGCNA_module_eigengenes.html">Identificaiton of Time- &amp; Colony-dependent modules</a></li>
<li class="toctree-l1"><a class="reference internal" href="RNAseq/WGCNA_module_eigengene_environment.html">Correlation of modules with environmental paramters</a></li>
<li class="toctree-l1"><a class="reference internal" href="RNAseq/CrossSpeciesNetwork.html">Construct cross-species network</a></li>
<li class="toctree-l1"><a class="reference internal" href="RNAseq/GeneSetAnalysis.html">Gene set analysis for the WGCNA modules</a></li>
</ul>
</div>
</div>
<div class="section" id="differential-expression-analysis">
<h3>Differential expression analysis<a class="headerlink" href="#differential-expression-analysis" title="Permalink to this headline">¶</a></h3>
<p>Identification of genes differentially expressed in the colonies harboring Endozoicomonas</p>
<div class="toctree-wrapper compound">
<ul>
<li class="toctree-l1"><a class="reference internal" href="RNAseq/Differential_expression_analysis.html">Differential expression analysis</a></li>
</ul>
</div>
</div>
</div>
</div>


           </div>
          </div>
          <footer>
  
    <div class="rst-footer-buttons" role="navigation" aria-label="footer navigation">
      
        <a href="RNAseq/Contig_taxonomy.html" class="btn btn-neutral float-right" title="Taxonomic assignments of the contigs" accesskey="n">Next <span class="fa fa-arrow-circle-right"></span></a>
      
      
        <a href="16S/UniFrac.html" class="btn btn-neutral" title="UniFrac analysis" accesskey="p"><span class="fa fa-arrow-circle-left"></span> Previous</a>
      
    </div>
  

  <hr/>

  <div role="contentinfo">
    <p>
        &copy; Copyright 2020, Toru Maruyama.

    </p>
  </div>
  Built with <a href="http://sphinx-doc.org/">Sphinx</a> using a <a href="https://github.com/snide/sphinx_rtd_theme">theme</a> provided by <a href="https://readthedocs.org">Read the Docs</a>. 

</footer>

        </div>
      </div>

    </section>

  </div>
  


  

    <script type="text/javascript">
        var DOCUMENTATION_OPTIONS = {
            URL_ROOT:'./',
            VERSION:'1.0',
            COLLAPSE_INDEX:false,
            FILE_SUFFIX:'.html',
            HAS_SOURCE:  true
        };
    </script>
      <script type="text/javascript" src="_static/jquery.js"></script>
      <script type="text/javascript" src="_static/underscore.js"></script>
      <script type="text/javascript" src="_static/doctools.js"></script>

  

  
  
    <script type="text/javascript" src="_static/js/theme.js"></script>
  

  
  
  <script type="text/javascript">
      jQuery(function () {
          SphinxRtdTheme.StickyNav.enable();
      });
  </script>
   

</body>
</html>